portality/view/doaj.py

```diff
@@ -341,8 +341,10 @@ def toc(identifier=None):
 
 
 @blueprint.route("/toc/articles/<identifier>")
-@blueprint.route("/toc/articles/<identifier>/<volume>")
-@blueprint.route("/toc/articles/<identifier>/<volume>/<issue>")
+def toc_articles_legacy(identifier=None, volume=None, issue=None):
+    return redirect(url_for('doaj.toc_articles', identifier=identifier, volume=volume, issue=issue), 301)
+
+@blueprint.route("/toc/<identifier>/articles")
 def toc_articles(identifier=None, volume=None, issue=None):
```

Comment:
better to remove parameter `volume, issue` and following code since `toc_articles` no longer receive `volume` and `issue` 
```
        if is_issn_by_identifier(identifier) and volume:
            filters = [dao.Facetview2.make_term_filter('bibjson.journal.volume.exact', volume)]
            if issue:
                filters += [dao.Facetview2.make_term_filter('bibjson.journal.number.exact', issue)]
            q = dao.Facetview2.make_query(filters=filters)

            # The issn we are using to build the TOC
            issn = bibjson.get_preferred_issn()
            return redirect(url_for('doaj.toc', identifier=issn)
                            + '?source=' + dao.Facetview2.url_encode_query(q))
```